### DISCUSSION TITLE
Detail metrics are displayed on related entities views

### DISCUSSION THREAD

**Discussion by chimit** [laravel/nova-issues#5351] [Q&A]
- Laravel Version: 9.51.0
- Nova Version: 4.22.0
- PHP Version: 8.1.9
- Database Driver & Version:
- Operating System and Version: macOS
- Browser type and version: Safari
- Reproduction Repository: https://github.com/chimit/nova

### Description:

My Building resource has a metric that should be displayed only on the detail page. It's a dummy example:

```php
class Building extends Resource
{
    public function cards(NovaRequest $request)
    {
        return [
            (new Metrics\BuildingsPerActive())->onlyOnDetail(),
        ];
    }
```

But now this metric is also displayed on any Building related resources as well:

<img width="687" alt="image" src="https://user-images.githubusercontent.com/839349/217729599-1528fdc9-4049-4300-ad05-ea3584241c50.png">

Looks like a similar bug to [this one](https://github.com/laravel/nova-issues/issues/5245).

**Comment by crynobone** [comment]
I don't think this justifies it as a bug, the previous issue was because it can potentially clash with parent resource's related relationship (`Action Events` in the example). 

Feel free to submit a change request via <https://github.com/laravel/nova-issues/discussions/new?category=ideas-feature-requests> if you really want this to be changed).

**Comment by chimit** [comment]
@crynobone @davidhemphill, I tried to simulate quite a realistic situation when multiple metrics on the same page look misleading. Please, check my latest update on https://github.com/chimit/nova

In my example, both Building and Room in that building have renovation dates. They use the same metric (DRY). And in the UI they are hardly distinguishable and confusing. Mostly it's because metrics are displayed above the entity titles. They should at least be shown below.

The second problem is that both these metrics use the same cache and the one which loads later uses the wrong cache. On my example below the second number should have been 88 (based on room `renovated_at` date, not building `renovated_at`):

<img width="717" alt="image" src="https://user-images.githubusercontent.com/839349/221114417-d20f14de-9a75-4cd9-aedc-52ff22ff8cd1.png">

Please, reopen the issue.

**Comment by crynobone** [comment]
The root cause of your issue is that you are reusing the same Metric with caching without adding a unique identifier (custom `uriKey` etc).

**Comment by chimit** [comment]
But this identifier won't help to solve the problem because it's the exact same metric.

**Comment by thomasowow** [comment]
Currently I am having a similar situation, but instead I do not want the metric to be shown when in the context of a relationship. @chimit how did you solve your problem?
